src/hip_memory.cpp

```diff
@@ -1542,44 +1542,73 @@ void ihipMemsetKernel(hipStream_t stream, T* ptr, T val, size_t sizeBytes) {
                        sizeBytes, std::move(val));
 }
 
-
-// TODO-sync: function is async unless target is pinned host memory - then these are fully sync.
-hipError_t hipMemsetAsync(void* dst, int value, size_t sizeBytes, hipStream_t stream) {
-    HIP_INIT_SPECIAL_API((TRACE_MCMD), dst, value, sizeBytes, stream);
-
+typedef enum ihipMemsetCopyDataType {
```

Comment:
ihipMemsetCopyDataType seems a very odd name. Isn't ihipMemsetDataType sufficient? In case you have plans to share this across memcpy routines as well, maybe we can use ihipMemOpDataType or something like that.

Reply:
I can change it to ihipMemsetDataType.

Reply:
Changing it to ihipMemsetDataType will be better